Add license information

joshuagl opened the pull request:
<!--
Thanks for opening a pull request! Please do not just delete this text.  The three fields below are mandatory.

Please remember to:
- This PR requires an issue. If it is a new feature, the issue should proceed the PR and will have allowed sufficent time for discussions to take place. Pleases use
issue tags such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ".
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!

Thank you :)
-->

#### Summary
<!--
 Explain the **motivation** for making this change. What existing problem does the pull request solve? How can reviewers test this PR?
-->

root-signing does not include any licensing information. The rest of sigstore is Apache-2.0 licensed. Assuming the same is true here, this PR adds a LICENSE file and adds SPDX license headers to all code files.
The copyright year for the license headers has been set to the year in which the file was first created.

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to an administrator running private sigstore instances (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense.

-->

NONE

#### Documentation
<!--

Does this change require an update to documentation? How will users implement your new feature?

Please reference a PR within https://docs.sigstore.dev

-->

joshuagl committed:
Remove second, spurious, emeritus section

We already have an "Emeritus Keyholders" section in the README,
remove the second (empty) one.

Signed-off-by: Joshua Lock <jlock@vmware.com>

joshuagl committed:
Add LICENSE file and license headers to source code

Add an Apache-2.0 LICENSE file to the root of the repository and mark all
code files with SPDX license headers where the copyright year is the year
in which the file was first created.

Signed-off-by: Joshua Lock <jlock@vmware.com>

asraa reviewed (commented):
Thank you so much!

While you're at it, could you add a COPYRIGHT.txt as well? I guess that's missing too
https://github.com/sigstore/sigstore/blob/main/COPYRIGHT.txt

joshuagl committed:
Add explicit copyright notice

Adds COPYRIGHT.txt, copied from from sigstore/sigstore
https://github.com/sigstore/sigstore/blob/31689aabfac5d1baffbc9ce5b38ede24e7029e44/COPYRIGHT.txt

Signed-off-by: Joshua Lock <jlock@vmware.com>

joshuagl commented:
> While you're at it, could you add a COPYRIGHT.txt as well? I guess that's missing too https://github.com/sigstore/sigstore/blob/main/COPYRIGHT.txt

I've pushed an additional commit with the COPYRIGHT.txt from sigstore/sigstore

Hayden-IO commented on COPYRIGHT.txt:
```diff
@@ -0,0 +1,14 @@
+
+Copyright 2021 The Sigstore Authors.
```
2022

asraa replied:
> The copyright year for the license headers has been set to the year in which the file was first created.

I assumed

asraa replied:
Ah, I guess this one can be 2022? I'm not sure. I assumed it was just "when was the dir first created"

Hayden-IO replied:
I’d probably go with this year, I don’t know if you can backdate copyright.

joshuagl replied:
Perhaps we should just remove the year? https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/

Hayden-IO commented on cmd/tuf/app/add-delegation.go:
```diff
@@ -1,3 +1,18 @@
+//
```
2022 (and all other files)
